File: Week1 - C/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    int height, i, j;
    const int LOWERLIMIT = 1, UPPERLIMIT = 8, GAP = 2;

    do
    {
        //prompt the height of the pyramid
        height = get_int("Height");
    }

    //check whether the input is between the range
    while (height < LOWERLIMIT || height > UPPERLIMIT);

    //Loop for Columns
    for (i = 0; i < height; i++)
    {
        //Loop for left side of the pyramid
        for (j = 0; j < height; j++)
        {
            if (j >= height - i - 1)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        //Loop for the gap of the pyramid
        for (j = 0; j < GAP; j++)
        {
            printf(" ");
        }
        //Loop for right side of the pyramid
        for (j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        //new line
        printf("\n");
    }

}

File: Week6 - Python/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

gap = 2

# Prompt the height until the input is valid, between 1 and 8
while(True):
    Height_number = get_int("Height: ")

    if Height_number >= 1 and Height_number <= 8:
        break

# Loop acording to the pyramid's height
for i in range(Height_number):
    # Loop that prints the hashes
    for j in range(0, Height_number + gap + i + 1, 1):
        if j + i >= Height_number - 1 and not(j <= Height_number + gap - 1 and j >= Height_number):
            print("#", end="")
        else:
            print(" ", end="")
    print("")